libs/api/domains/cms/src/lib/models/url.model.ts

```diff
@@ -0,0 +1,64 @@
+import { Field, ObjectType, ID, createUnionType } from '@nestjs/graphql'
+
+import {
+  IArticle,
+  IArticleCategory,
+  ILifeEventPage,
+  INews,
+  IUrl,
+} from '../generated/contentfulTypes'
+
+import { News } from './news.model'
+import { Article } from './article.model'
+import { LifeEventPage } from './lifeEventPage.model'
+import { ArticleCategory } from './articleCategory.model'
+
+type UrlPageTypes = IArticle | IArticleCategory | INews | ILifeEventPage
+
+export const mapPage = ({ fields, sys }: UrlPageTypes) => ({
+  id: sys.id,
+  contentType: sys.contentType.sys.id,
+  title: fields.title,
+  slug: fields.slug,
+})
+
+export const UrlPage = createUnionType({
+  name: 'UrlPage',
+  types: () => [Article, ArticleCategory, News, LifeEventPage],
+  resolveType: (value) => {
```

Comment:
Other union types are solving this by having the union type mapper function return an instance of the class itself. Not sure if we should be using different methods of doing this for different union types